workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/components/LicenseSelector/index.tsx

```diff
@@ -39,19 +39,24 @@ export function LicenseSelector({
 	selectedPaymentMethod,
 	selectedProduct,
 	setLicenseSelected,
-	sku,
 }: LicenseSelectorProps) {
 	const [selectedTimeline, setSelectedTimeline] = useState('');
 	const [trialSKU, setTrialSKU] = useState<SKU>();
 	const [disabledButton, setDisabledButton] = useState<boolean>(false);
 
 	const hasTrialSkuVerification = useCallback(() => {
-		sku.skuOptions.forEach((option) => {
-			if (option.key === 'trial' && option.value === 'yes') {
-				setTrialSKU(sku);
-			}
-		});
-	}, [sku]);
+		const skus = selectedProduct?.skus;
+
+		const [isTrial] =
+			skus?.filter((sku) =>
+				sku?.skuOptions.find(
+					(skuOption) =>
+						skuOption?.key === 'trial' && skuOption?.value === 'yes'
+				)
+			) || [];
```

Comment:
Maybe using `some` is better than filtering here, since we need a boolean

```suggestion
		const isTrial =
			skus?.some((sku) =>
				sku?.skuOptions.find(
					(skuOption) =>
						skuOption?.key === 'trial' && skuOption?.value === 'yes'
				)
			);
```

Reply:
Hey Keven, actually we need the information of this trial option. Do you think that changing the variable name can help to clarify this?